Is `-DLLDB_TABLEGEN_EXE=/path/to/build/machine/lldb-tblgen` now required or recommended for cross-compiling LLVM?

Hi, after this PR,

- https://github.com/llvm/llvm-project/pull/159812

in our downstream cross-compilation of LLVM for the Android 7 command line, we have started experiencing this error during the final installation step:

```
"/home/builder/.termux-build/libllvm/build/bin/lldb-tblgen": No such file or directory
```

unless we add the `-DLLDB_TABLEGEN_EXE=/path/to/build/machine/lldb-tblgen` argument to the main CMake configuration command, which in our case is represented by `-DLLDB_TABLEGEN_EXE=$TERMUX_PKG_HOSTBUILD_DIR/bin/lldb-tblgen`, revert that PR using a patch, or add the `make lldb-tblgen` command in addition to the normal `make` command to the compilation step.

This is our work in progress build script for LLVM 22 that reproduces the error currently (link is to a version that does not have any of the three mentioned mitigations I found applied): https://github.com/termux/termux-packages/blob/2546529cc338c3273ddeb8ea9eac754c610c69bd/packages/libllvm/build.sh

I was wondering whether by asking here, I might be able to find out one or more of these things:

- What properties does the cross-compilation setup of upstream LLVM have that make that PR helpful for upstream?
- Are there any obvious differences between Termux's LLVM cross-compilation configuration and upstream's that should be corrected by changing Termux's configuration to be more like upstream's?
- Which of the three workarounds I have found, or none of them, seems like the best choice for a downstream project that is encountering an issue with the PR linked above?

I can reproduce the issue "cross compiling" to my own system using https://lldb.llvm.org/resources/build.html#example-2-cross-compiling-for-linux-arm64-on-ubuntu-host-using-an-existing-host-build and using `LLVM_NATIVE_TOOL_DIR` like you are doing.

After the build I have:
```
$ ls bin | grep tblgen
clang-tblgen
llvm-tblgen
```
Same install error:
```
CMake Error at tools/lldb/utils/TableGen/cmake_install.cmake:52 (file):
  file INSTALL cannot find "/home/davspi01/build-llvm-cross/bin/lldb-tblgen":
  No such file or directory.
```
And `ninja lldb-tblgen` "fixes" it.

If we are installing `lldb-tblgen` in native builds to make them more useful for cross builds, I don't see why we shouldn't be building the cross `lldb-tblgen` in the cross build to make it useful in the same way.

There is some extra code around `lldb-tblgen` claiming to be to do with standalone builds (building only lldb by using an existing llvm/clang). This may be the cause of it.

--- a/lldb/utils/TableGen/CMakeLists.txt
+++ b/lldb/utils/TableGen/CMakeLists.txt
@@ -9,6 +9,7 @@ if (NOT DEFINED LLDB_TABLEGEN_EXE)
 
     add_tablegen(lldb-tblgen LLDB
       DESTINATION "${CMAKE_INSTALL_BINDIR}"
+      EXPORT LLDB
       LLDBOptionDefEmitter.cpp
       LLDBPropertyDefEmitter.cpp
       LLDBTableGen.cpp

